### PR TITLE
fix(SuiviClasse) : EVAL-453 Fixes averages PDF of Suivi Classe encountering an error if a group tried to do the export.

### PR DESCRIPTION
**Pull request by FlorentMr**
## Describe your changes
Fixes averages PDF of Suivi Classe encountering an error if a group tried to do the export.

## Checklist tests
Suivi Classe -> Try to export a PDF in the "Moyennes" tab with a class and then with a group.

## Issue ticket number and link
[EVAL-453](https://jira.support-ent.fr/browse/EVAL-453)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by JinDouang — APPROVED** [review]

File: src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java [hunk 1]
```diff
@@ -1984,15 +1984,23 @@ public void handle(Either<String, JsonArray> responseQuerry) {
                     handler.handle(new Either.Left<>(error));
                 } else {
                     JsonArray idClasseGroups = responseQuerry.right().getValue();
-                    if (idClasseGroups == null || idClasseGroups.isEmpty()) {
+                    if ((idClasseGroups == null || idClasseGroups.isEmpty()) && typeGroupe == 0) {
                         handler.handle(new Either.Left<>("idClasseGroups null or empty"));
                     } else {
                         JsonArray idsGroups = new fr.wseduc.webutils.collections.JsonArray();
+                        final String nameClasse, idClass;
```
Comment (BastienBenard) [inline src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java:1992]: Tu as oublié de changer le message de l'Either
Pour évité d'avoir des pyramide de if tu peux utiliser des return dans tes if et ne pas utiliser de else. 
```
                    JsonArray idClasseGroups = responseQuerry.right().getValue();
                    if (idClasseGroups == null || idClasseGroups.isEmpty()) {
                        handler.handle(new Either.Left<>("idClasseGroups null or empty"));
                        return;
                    }
                    if (typeGroupe == 0) {
                        handler.handle(new Either.Left<>("typeGroup is 0"));
                        return;
                    }

                    JsonArray idsGroups = new fr.wseduc.webutils.collections.JsonArray();
                    final String nameClasse, idClass;
```

File: src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java [hunk 2]
```diff
@@ -1984,15 +1984,23 @@ public void handle(Either<String, JsonArray> responseQuerry) {
...
 
-                        idsGroups.add(idClasseGroups.getJsonObject(0).getString(Field.ID_CLASSE));
-                        idsGroups.addAll(idClasseGroups.getJsonObject(0).getJsonArray("id_groupes"));
-                        final String nameClasse = idClasseGroups.getJsonObject(0).getString("name_classe");
-                        final String idClass = idClasseGroups.getJsonObject(0).getString(Field.ID_CLASSE);
+                        if(typeGroupe != 0){
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java:1993]: cette condition est toujours vrai, non ?

File: src/main/java/fr/openent/competences/controllers/ExportPDFController.java [hunk 3]
```diff
@@ -1562,17 +1562,19 @@ public void handle(UserInfos eventUser) {
                 if (eventUser == null) {
                     unauthorized(request);
                 } else {
-                    String idClasse = request.params().get("idClasse");
-                    String idEtablissement = request.params().get("idEtablissement");
+                    String idClasse = request.params().get(Field.IDCLASSE);
+                    String idEtablissement = request.params().get(Field.IDETABLISSEMENT);
+                    final Integer typeGroupe = Integer.valueOf(request.params().get(Field.TYPEGROUPE));
```
Comment (BastienBenard) [inline src/main/java/fr/openent/competences/controllers/ExportPDFController.java:1567]: Pour le nommage de t'es variable typeGroupe -> typeGroup

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java (hunk 2 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java:1993]: nop si idClasseGroups contient 1 élément et que typeGroupe est a un 1 par exemple

**Review by lapeyronnie — APPROVED** [review]

**Pull request merged**